### ISSUE TITLE
Question: How to properly encode a JEXL Geo:Point

### ISSUE THREAD

**Issue by jason-fox** [telefonicaid/iotagent-node-lib#1150]
I've been looking at [the documentation](https://iotagent-node-lib.readthedocs.io/en/latest/expressionLanguage/index.html#jexl-based-transformations), and there is no complete working example of a GeoJSON `Point` encoding for the JEXL parser:

I currently have the following:

```json
{
    "devices": [
        {
            "device_id": "motion010",
            "entity_name": "urn:ngsi-ld:Motion:010",
            "entity_type": "Motion",
            "timezone": "Europe/Berlin",
            "expressionLanguage": "jexl",
            "attributes": [
                {
                    "name": "location",
                    "type": "geo:point",
                    "expression": "longitude + \",\" + latitude"
                },
                {
                    "name": "TimeInstant",
                    "type": "DateTime",
                    "expression": "level|toisodate"
                },
                {
                    "name": "level",
                    "type": "Number"
                },
                {
                    "name": "latitude",
                    "type": "Number"
                },
                {
                    "name": "longitude",
                    "type": "Number"
                }
            ]
        }
    ]
}
```

Which results in the following:

```json
{
    "id": "urn:ngsi-ld:Motion:010",
    "type": "Motion",
    "TimeInstant": {
        "type": "DateTime",
        "value": "2021-11-18T14:20:14.901Z",
        "metadata": {}
    },
    "latitude": {
        "type": "Number",
        "value": 1,
        "metadata": {
            "TimeInstant": {
                "type": "DateTime",
                "value": "2021-11-18T14:20:14.901Z"
            }
        }
    },
    "level": {
        "type": "Number",
        "value": 1637245214901,
        "metadata": {
            "TimeInstant": {
                "type": "DateTime",
                "value": "2021-11-18T14:20:14.901Z"
            }
        }
    },
    "location": {
        "type": "geo:point",
        "value": "7,1",
        "metadata": {
            "TimeInstant": {
                "type": "DateTime",
                "value": "2021-11-18T14:20:14.901Z"
            }
        }
    },
    "longitude": {
        "type": "Number",
        "value": 7,
        "metadata": {
            "TimeInstant": {
                "type": "DateTime",
                "value": "2021-11-18T14:20:14.901Z"
            }
        }
    }
}
```

But instead of:

```json
"type": "geo:point", "value": "7,1"
```

 I would like to have

```json
"type": "geo:point", "value": {"coordinates": [7,1], "type": "Point"}
```

and I can't see how to encode the quotation marks  within the provisioning statement. The closest I can get in the [JEXL Playground](https://czosel.github.io/jexl-playground/#/?context=%7B%0A%20%20%22longitude%22%3A%205%2C%0A%20%20%22latitude%22%3A%2036%0A%7D&input=%22%7B'coordinates'%3A%20%5B%22%20%2B%20.longitude%20%0A%2B%20%22%2C%22%20%2B%20.latitude%20%2B%20%22%5D%2C%20'type'%3A%20'Point'%7D%22%20%20%20) is single quote which won't work with a `jsonparse` function.

**Comment by jason-fox** [comment]
I've had a thought - maybe the mandatory GeoJSON conversions defined for NGSI-LD could be exposed within the JEXL parser, resulting in an expression like:

```json
"expression": "longitude + \",\" + latitude | toGeoJSON"
```

That would make #1020 obsolete.

**Comment by mrutid** [comment]
Try this [PLAYGROUND](https://czosel.github.io/jexl-playground/#/?context=%7B%0A%20%20%22longitude%22%3A%205%2C%0A%20%20%22latitude%22%3A%2036%0A%7D&input=%7Bcoordinates%3A%20%5Blongitude%2Clatitude%5D%2C%20type%3A%20'Point'%7D)

`
expression: "{coordinates: [longitude,latitude], type: 'Point'}"
`
Every Object auto-map to JSON at return.

I Agree. Now finally JEXL is able to control explicitly every casting needed. But do we want to force to use JEXT to every user?

**Comment by mapedraza** [comment]
Having into account @mrutid answer, can we close this issue?

**Comment by jason-fox** [comment]
The question is answered, but personally I'd update the documentation with PR #1152 **first** before closing, but that is just purely a personal preference. Much better to have all this information in the public documentation than just an closed issue IMHO.

The real question (for me) is whether Telefónica agree JEXL parsing properly implemented is sufficient to close #1020 unmerged - my thought here is that #1020 becomes unnecessary (and should be withdrawn)  if there is sufficient documentation to cover the workaround using JEXL there is no need for an additional flag just to be able to do the same thing using the legacy parser.

**Comment by fgalan** [comment]
> The question is answered, but personally I'd update the documentation with PR #1152 first before closing, but that is just purely a personal preference. Much better to have all this information in the public documentation than just an closed issue IMHO.

We can keep this issue opened until PR #1052 gets merged.
